homeassistant/components/subaru/sensor.py

```diff
@@ -0,0 +1,242 @@
+                subaru_sensor[SENSOR_FIELD],
+                subaru_sensor[SENSOR_UNITS],
+            )
+        )
+
+
+class SubaruSensor(SubaruEntity):
+    """Class for Subaru sensors."""
+
+    def __init__(self, vehicle_info, coordinator, title, data_field, api_unit):
+        """Initialize the sensor."""
+        super().__init__(vehicle_info, coordinator)
+        self.hass_type = "sensor"
+        self.current_value = None
+        self.title = title
+        self.data_field = data_field
+        self.api_unit = api_unit
+
+    @property
+    def icon(self):
+        """Return icon for sensor."""
+        if self.title == "EV Battery Level" and self.coordinator.data.get(self.vin):
+            charge_status = (
+                self.coordinator.data[self.vin]["status"].get(sc.EV_CHARGER_STATE_TYPE)
+                == "CHARGING"
+            )
+            return icon_for_battery_level(
+                battery_level=self.current_value, charging=charge_status
+            )
+        return super().icon
+
+    @property
+    def state(self):
+        """Return the state of the sensor."""
+        self.current_value = self.get_current_value()
+
+        if self.current_value is None:
+            return None
+
+        if self.api_unit in TEMPERATURE_UNITS:
+            return round(
+                self.hass.config.units.temperature(self.current_value, self.api_unit), 1
+            )
+
+        if self.api_unit in LENGTH_UNITS:
+            return round(
+                self.hass.config.units.length(self.current_value, self.api_unit), 1
+            )
+
+        if self.api_unit in PRESSURE_UNITS:
+            if self.hass.config.units == IMPERIAL_SYSTEM:
+                return round(
+                    self.hass.config.units.pressure(self.current_value, self.api_unit),
+                    1,
+                )
+
+        if self.api_unit in FUEL_CONSUMPTION_UNITS:
+            if self.hass.config.units == IMPERIAL_SYSTEM:
+                return round((100.0 * L_PER_GAL) / (KM_PER_MI * self.current_value), 1)
+
+        return self.current_value
+
+    @property
+    def unit_of_measurement(self):
+        """Return the unit_of_measurement of the device."""
+        if self.api_unit in TEMPERATURE_UNITS:
+            return self.hass.config.units.temperature_unit
+
+        if self.api_unit in LENGTH_UNITS:
+            return self.hass.config.units.length_unit
+
+        if self.api_unit in PRESSURE_UNITS:
+            if self.hass.config.units == IMPERIAL_SYSTEM:
+                return self.hass.config.units.pressure_unit
+            return PRESSURE_HPA
+
+        if self.api_unit in FUEL_CONSUMPTION_UNITS:
+            if self.hass.config.units == IMPERIAL_SYSTEM:
+                return FUEL_CONSUMPTION_MPG
+            return FUEL_CONSUMPTION_L_PER_100KM
+
+        return self.api_unit
+
+    def get_current_value(self):
+        """Get raw value from the coordinator."""
+        if self.coordinator.data.get(self.vin):
```

Comment:
```suggestion
        if self.vin not in self.coordinator.data:
            return None
```

Then outdent the block below

Reply:
I just removed this check. With new available property added, I was unable to create a test case that would reach this line.